src/PAModel/SourceTransforms/ComponentInstanceTransform.cs

```diff
@@ -37,12 +37,14 @@ public void BeforeWrite(BlockNode control)
         private void DoRename(BlockNode control)
         {
             var templateName = control.Name?.Kind?.TypeName ?? string.Empty;
-            if (!ComponentRenames.TryGetValue(templateName, out var rename))
+            if (ComponentRenames.TryGetValue(templateName, out var rename))
+            {
+                control.Name.Kind.TypeName = rename;
+            }
+            else
             {
                 _errors.ValidationError("Renaming component instance but unable to find target name");
```

Comment:
This would still records validation errors(PA3005) I believe. Probably need to handle those too.

Reply:
We want it to record an error, we just don't want it throwing the exception like it had before (throw new DocumentException();)